docker/init-db.sh

```diff
@@ -8,7 +8,7 @@
 set -eu
 
 psql -v ON_ERROR_STOP=1 --username "${POSTGRES_USER}" > /dev/null <<-EOSQL
-  CREATE USER ${MARQUEZ_USER};
+  CREATE USER ${MARQUEZ_USER} SUPERUSER;
```

Comment:
@wslulciuc @collado-mike @mobuchowski This line is worth looking at and perhaps a discussion. 

As long we're using PSQL 12, super user is required to install `uuid-ossp` extension. We can remove it once upgrading PostgreSQL version to 13. 

Additionally, the migration code should work in a way such that it is allowed to run externally sql:
```
CREATE EXTENSION IF NOT EXISTS "uuid-ossp"
```
and run migration with no extra privileges. 

The CHANGELOG info about this will be included in the PR3.

Reply:
It'd be great to [upgrade](https://github.com/MarquezProject/marquez/issues/2364) our postgres version, but we'll want to decouple the upgrade from the migration. Meaning, there's no way arounds having to require users to run the migration as `superuser`. Since requiring `superuser` is only for the migration, providing migration steps in the changelog would be enough.

Reply:
Hmm, requiring the Postgres upgrade and/or running the migration process as a superuser seems like a serious hurdle. What's the principal gain the user gets from this? I.e., why do the facet tables require a UUID primary key to begin with? AFAIK, we only join with those tables based on the foreign keys (run_uuid, job_uuid, dataset_uuid). If we get rid of the primary keys on those facet records, we don't have to worry about new extensions or postgres upgrades and I don't know if there's any downside to eliminating them.

Reply:
Uuid as a primary key has been proposed in proposal #2076 some time ago before I started working on that.
For me it looked like some general pattern in Marquez, that we create primary keys for tables so that database schema can be easily extended in future (easy joining of tables). For these tables, primary key cannot be derived from other columns. But I don't see any blocker in removing those columns.

Reply:
Unfortunately, yes the migration would be a hurdle for users running Marquez with postgres `12` or below. As @pawel-big-lebowski mentioned, we do have a general pattern for defining a `uuid` (=**PK**) column for all tables (mapping tables excluded ex: `dataset_fields_tag_mapping`). It would be great to keep this convention as we don't know how the facet tables will be used or extended. But, outside breaking convention, I don't see a major downside.

Reply:
We can still append a new facet for the same run even without the UUID. The primary key is principally useful for fetching a specific record or referencing a specific record from another table. Neither of those is the case here. 

Arguing that we don't know how the facet tables will be used doesn't make sense either, as we can always add the UUID column after the fact. When there's a reason to introduce the primary key, we can argue that the extra work is worth the effort. For now, the current migration requires a very substantial change in the system with no benefit. Following the pattern simply for the sake of the pattern isn't a good reason to require these changes to the system or the deployment process that users are following.